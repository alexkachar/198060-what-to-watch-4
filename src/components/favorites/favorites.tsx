import * as React from 'react';
import {connect} from 'react-redux';
import {Redirect} from 'react-router-dom';

import DataOperation from '../../store/operations/data/data';
import {getFavorites} from '../../store/reducers/data/selectors';
import {getAuthFlag} from '../../store/reducers/user/selectors';
import {AppRoutes} from '../../constants';

import SvgInjection from '../partials/svg-injection/svg-injection';
import Header from '../header/header';
import Footer from '../footer/footer';
import MovieCard from '../movie-card/movie-card';
import Movie from '../../interfaces/movie';

interface Props {
  onRequestFavorites: () => void;
  favorites: Movie[];
  isAuth: boolean;
}

class Favorites extends React.PureComponent<Props> {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    const {onRequestFavorites} = this.props;
    onRequestFavorites();
  }

  render() {
    if (!this.props.isAuth) {
      return <Redirect to={AppRoutes.MAIN} />;
    }

    const {favorites} = this.props;
    return (
      <>
        <SvgInjection />

        <div className="user-page">

          <Header isFavoritesHeader={true} />
          <section className="catalog">
            <h2 className="catalog__title visually-hidden">Catalog</h2>

            {favorites.map((movie) => <MovieCard movie={movie} key={movie.id} />)}

          </section>
          <Footer />
        </div>
      </>
    );
  }
}

const mapStateToProps = (state) => ({
  favorites: getFavorites(state),
  isAuth: getAuthFlag(state)
});

const mapDispatchToProps = (dispatch) => ({
  onRequestFavorites: () => {
    dispatch(DataOperation.loadFavorites());
  }
});

export {Favorites};
export default connect(mapStateToProps, mapDispatchToProps)(Favorites);
