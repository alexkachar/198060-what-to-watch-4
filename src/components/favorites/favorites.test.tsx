import * as React from 'react';
import * as renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';

import {Provider} from 'react-redux';
import {Favorites} from './favorites';
import MOCK_STORE from '../../test-data/mock-store';
import MOCK_MOVIES from '../../test-data/mock-movies';
import Movie from '../../interfaces/movie';

const FAVORITES: Movie[] = MOCK_MOVIES;

it(`MoviePage renders correctly`, () => {
  const tree = renderer
    .create(
        <Provider store={MOCK_STORE}>
          <BrowserRouter>
            <Favorites
              favorites={FAVORITES}
              isAuth={true}
              onRequestFavorites={jest.fn()} />
          </BrowserRouter>
        </Provider>, {
          createNodeMock: () => document.createElement(`div`)
        })
    .toJSON();
  expect(tree).toMatchSnapshot();
});
