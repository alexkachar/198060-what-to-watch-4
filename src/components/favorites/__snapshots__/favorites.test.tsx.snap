// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviePage renders correctly 1`] = `
<div
  className="user-page"
>
  <h1
    className="visually-hidden"
  >
    WTW
  </h1>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="page-title user-page__title"
    >
      My list
    </h1>
    <div
      className="user-block"
    >
      <a
        className="user-block__avatar"
        href="/favorites"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <img
          src="https://htmlacademy-react-3.appspot.com/img/john_doe.jpg"
          style={
            Object {
              "borderRadius": "50%",
              "height": "63px",
              "width": "63px",
            }
          }
        />
      </a>
    </div>
  </header>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <div
      className="catalog__movies-list"
    >
      <article
        className="small-movie-card catalog__movies-card"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <a
          href="/film/1"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height="175"
              muted={true}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              poster="img/the-grand-budapest-hotel.jpg"
              src="https://some-link"
              width="280"
            />
          </div>
        </a>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/film/1"
            onClick={[Function]}
            style={
              Object {
                "color": "#c9b37e",
              }
            }
          >
            The Grand Budapest Hotel
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <a
          href="/film/2"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height="175"
              muted={true}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              poster="https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/moonrise-kingdom.jpg"
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              width="280"
            />
          </div>
        </a>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/film/2"
            onClick={[Function]}
            style={
              Object {
                "color": "#c9b37e",
              }
            }
          >
            Moonrise Kingdom
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <a
          href="/film/15"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height="175"
              muted={true}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              poster="https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/fantastic-beasts-the-crimes-of-grindelwald.jpg"
              src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
              width="280"
            />
          </div>
        </a>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/film/15"
            onClick={[Function]}
            style={
              Object {
                "color": "#c9b37e",
              }
            }
          >
            Fantastic Beasts: The Crimes of Grindelwald
          </a>
        </h3>
      </article>
    </div>
  </section>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        Â© 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
